tests/js/server/aql/aql-optimizer-rule-optimize-cluster-single-document-operations.js

```diff
@@ -342,9 +349,13 @@ function optimizerClusterSingleDocumentTestSuite () {
         [ "UPDATE {_key: '1'} WITH {foo: 'bar4a'} IN " + cn1 + " OPTIONS {} RETURN [OLD, NEW]", 2, 2, true, setupC1, 0],        
         [ "UPDATE {_key: '1'} WITH {foo: 'bar5a'} IN " + cn1 + " OPTIONS {} RETURN { old: OLD, new: NEW }", 2, 2, true, setupC1, 0],
         [ "UPDATE {_key: '1'} INTO " + cn1 + " RETURN OLD._key", 0, 2, true, s, 0],
+        [ "UPDATE {_key: '1'} WITH {_key: '1', name: 'test1'} IN " + cn1 + " OPTIONS {} RETURN NEW", 1, 1, true, setupC1, 0, "name", "test1"],
```

Comment:
For completeness, can you also add tests for
* updating with _only_ the `_key` in `WITH` set to the same value as the lookup key, i.e. `UPDATE {_key: '1'} INTO ... WITH {_key: '1'} IN ...` 
* updating with _only_ the `_key` in `WITH` set to a different value than the lookup key, ie. `UPDATE {_key: '1'} INTO ... WITH {_key: '2'} IN ...` 
* same as above, but using only the string lookup key, i.e. `UPDATE '1' INTO ... WITH {_key: '1'} IN ...` and `UPDATE '1' INTO ... WITH {_key: '2'} IN ...` 

Plus all of that for REPLACE as well.